pandas/core/groupby/generic.py

```diff
@@ -1069,13 +1069,37 @@ def nsmallest(
 
     @doc(Series.idxmin.__doc__)
     def idxmin(self, axis: Axis = 0, skipna: bool = True) -> Series:
-        result = self._op_via_apply("idxmin", axis=axis, skipna=skipna)
-        return result
+        axis = self.axis
+
+        def func(df):
+            return df.idxmin(axis=axis, skipna=skipna)
+
+        func.__name__ = "idxmin"
+        return self._idxmin_idxmax(func)
 
     @doc(Series.idxmax.__doc__)
     def idxmax(self, axis: Axis = 0, skipna: bool = True) -> Series:
-        result = self._op_via_apply("idxmax", axis=axis, skipna=skipna)
-        return result
+        axis = self.axis
+
+        def func(df):
+            return df.idxmax(axis=axis, skipna=skipna)
+
+        func.__name__ = "idxmax"
+        return self._idxmin_idxmax(func)
+
+    def _idxmin_idxmax(self, func: Callable) -> Series:
+        result, _ = self.grouper.apply(func, self._obj_with_exclusions, self.axis)
+
+        if len(result) == 0:
+            return Series(
+                [],
+                name=self._obj_with_exclusions.name,
+                index=self.grouper.result_index,
+            ).astype(self._obj_with_exclusions.index.dtype)
```

Comment:
can you just pass the dtype keyword to Series?  also probably needs to be self.obj._constructor